### PR TITLE
Mounting Google Drive with elFinder

### PR DESCRIPTION
**Pull request by rajasharma27**
Sir, 
after removing all unnecessary code.

```php
// ===============================================
/**
 * # Google Drive volume driver need "google-api-php-client"  latest version package"
 * * google-api-php-client: https://github.com/google/google-api-php-client
 * * copy directory "google-api-php-client" and paste in elFinder php directory
 * * Google Drive developer console: https://console.developers.google.com
 * * Create login in google developer console and create app and get clientid, clientsecret, redirecturi
 * * create new project ->Enable and manage APIs ->Drive API ->Enable ->Goto Credentials
 * * Create Credentials ->oAuth Client ->web Applicaton
 * */ 
 // Required for Google Drive Netmount

 include_once dirname(__FILE__).DIRECTORY_SEPARATOR.'elFinderVolumeGoogleDrive.class.php';

 // GoogleDrive Netmount driver need next two settings. You can get at https://console.developers.google.com
 // AND reuire regist redirect url to "YOUR_CONNECTOR_URL?cmd=netmount&protocol=googledrive&host=1" 
 define('ELFINDER_GOOGLEDRIVE_CLIENTID', 		'');
 define('ELFINDER_GOOGLEDRIVE_CLIENTSECRET',	'');
// ===============================================
```

### CODE REVIEW

**Comment by rajasharma27** [comment]
sir,
please add in your elFinder

**Comment by nao-pon** [comment]
@rajasharma27 It should not be disabled about SSL certificate checking. Please fix it.

**Comment by rajasharma27** [comment]
sir, 
remove done

**Comment by nao-pon** [comment]
@rajasharma27 Netmount dose not work.

**Comment by rajasharma27** [comment]
Sir,
if we remove disable SSL certificate checking
GuzzleHttp  require SSL  CA certificate
php.ini    curl.cainfo = "C:\windows\cacert.pem"

**Comment by nao-pon** [comment]
@rajasharma27 No, it is not match to [netmount option](https://github.com/Studio-42/elFinder/blob/master/js/elFinder.options.js#L358).

Other, I think this driver prerequisite is using composer. Therefore, "include_once 'google-api-php-client/vendor/autoload.php'" is unnecessary.

**Comment by rajasharma27** [comment]
little code comment in elFinder js file, this driver automatically find folder path like dropbox options

googledrive: {
				name : 'GoogleDrive.com',
				inputs: {
					//offline  : $('<input type="checkbox"/>').on('change', function() {
//						$(this).parents('table.elfinder-netmount-tb').find('select:first').trigger('change', 'reset');
//					}),
					host     : $('<span><span class="elfinder-info-spinner"/></span><input type="hidden"/>'),
					path     : $('<input type="text" value="/"/>'),  // change value root to /
					user     : $('<input type="hidden"/>'),
					pass     : $('<input type="hidden"/>')
				},
				select: function(fm, ev, data){
					var f = this.inputs, oline = f.offline,
						data = data || null;
					if ($(f.host[0]).find('span.elfinder-info-spinner').length || data === 'reset') {
						//if (oline.parent().children().length === 1) {
//							f.path.parent().prev().html(fm.i18n('folderId'));
//							oline.attr('title', fm.i18n('offlineAccess'));
//							oline.uniqueId().after($('<label/>').attr('for', oline.attr('id')).html(' '+fm.i18n('offlineAccess')));
//						}
						$(f.host[0]).empty().addClass('elfinder-info-spinner')
							.parent().find('span.elfinder-button-icon').remove();
						fm.request({
							data : {cmd : 'netmount', protocol: 'googledrive', host: 'googledrive.com', user: 'init', pass: 'init', options: {url: fm.uploadURL, id: fm.id}},  //add pass : 'init'
							preventDefault : true
						}).done(function(data){ 
							$(f.host[0]).removeClass("elfinder-info-spinner").html(data.body.replace(/\{msg:([^}]+)\}/g, function(whole,s1){return fm.i18n(s1,'GoogleDrive.com');}));
						}).fail(function(){});
					} else {
						oline.parent().parent()[f.user.val()? 'hide':'show']();
					}
				},
				done: function(fm, data){
					var f = this.inputs, p = this.protocol;
					
					if (data.mode == 'makebtn') {
						$(f.host[0]).removeClass('elfinder-info-spinner');
						f.host.find('input').hover(function(){$(this).toggleClass('ui-state-hover');});
						$(f.host[1]).val('');
						f.path.val('/').next().remove();    // change root to /
						f.user.val('');
						f.pass.val('');
						//f.offline.parent().parent().show();
					} else {
						$(f.host[0]).html('GoogleDrive.com&nbsp;').removeClass('elfinder-info-spinner');
						if (data.reset) {
							p.trigger('change', 'reset');
							return;
						}
						$(f.host[0]).parent().append($('<span class="elfinder-button-icon elfinder-button-icon-reload" title="'+fm.i18n('reAuth')+'">')
							.on('click', function() {
								$(f.host[1]).val('reauth');
								p.trigger('change', 'reset');
							}));
						$(f.host[1]).val('googledrive');
						if (data.folders) {
							f.path.next().remove().end().after(
								$('<div/>').append(
									$('<select class="ui-corner-all" style="max-width:200px;">').append(
										$($.map(data.folders, function(n,i){return '<option value="'+i+'">'+fm.escape(n)+'</option>'}).join(''))
									).on('change', function(){f.path.val($(this).val());})
								)
							);
						}
						f.user.val('done');
						f.pass.val('done');
						f.offline.parent().parent().hide();
					}
				},
				fail: function(fm, err){
					this.protocol.trigger('change', 'reset');
				}
			}
		},

**Comment by nao-pon** [comment]
I got some notice errors. Please check with `error_reporting(-1)` and fix it.

**Comment by nao-pon** [comment]
I think that `$client->setScopes([ Google_Service_Drive::DRIVE ])` is enough for access permissions.

**Comment by rajasharma27** [comment]
i am not getting

I got some notice errors. Please check with  error_reporting(-1)  and fix it.

P.S. Do not use the "@" as possible.

**Comment by rajasharma27** [comment]
ok 
found error

File: php/elFinderVolumeGoogleDrive.class.php [hunk 1]
```diff
@@ -234,7 +234,7 @@ public function netmountPrepare($options) {
 						$client->setClientId($options['client_id']);
 						$client->setClientSecret($options['client_secret']);
 						$client->setRedirectUri($this->getConnectorUrl().'?cmd=netmount&protocol=googledrive&host=1');						
-						$client->setScopes(array('https://www.googleapis.com/auth/userinfo.profile  https://www.googleapis.com/auth/drive https://www.googleapis.com/auth/drive.file https://www.googleapis.com/auth/drive.metadata'));
+						$client->setScopes([Google_Service_Oauth2::USERINFO_PROFILE , Google_Service_Drive::DRIVE]);
```
Comment (nao-pon) [inline php/elFinderVolumeGoogleDrive.class.php]: Is it need user id?

**Comment by rajasharma27** [comment]
P.S. Do not use the "@" as possible. ?

**Comment by rajasharma27** [comment]
sir,
I am not getting this
P.S. Do not use the "@" as possible

**Comment by nao-pon** [comment]
Sorry to my poor English. 

Please avoid error suppression using the '@'.

**Comment by rajasharma27** [comment]
PHP supports one error control operator: the at sign (@). When prepended to an expression in PHP, any error messages that might be generated by that expression will be ignored. 

thanks

**Comment by rajasharma27** [comment]
return error when uploading 20 or more files
 "error": {
  "code": 500,
  "message": null
 }
this error belong to php setting or drive class ?

**Comment by rajasharma27** [comment]
Sir,
kindly suggest me for any changes

**Comment by rajasharma27** [comment]
Sir,
only one point not clear
when check root folder info it showing  only current folder size not full drive
please suggest

**Comment by nao-pon** [comment]
Please full support of [`commandsOptions.netmount.googledrive`](https://github.com/Studio-42/elFinder/blob/master/js/elFinder.options.js#L359-L430).

**Comment by rajasharma27** [comment]
ok sir
I am changing code according to  commandsOptions.netmount.googledrive .

**Comment by rajasharma27** [comment]
Sir, 
this class working according to elfinder.
only one issue ( when check folder or root folder info) display size only current folder
please help

**Comment by rajasharma27** [comment]
thanks

**Pull request closed without merging**